### ISSUE TITLE
Redundant BatchNorm in create_head after Flatten

### ISSUE THREAD

**Issue by muellerzr** [fastai/fastai#3012]
`create_head` will utilize `LinBnDrop` when creating it's head. As a result we end up in a situation such as:
```python
BatchNorm2d -> AdaptiveAvgPool2d -> AdaptiveMaxPool2d -> Flatten -> BatchNorm1d
```
(A better visualization is below): 
![image](https://user-images.githubusercontent.com/7831895/99588600-d0477f80-29b8-11eb-97d9-a27a22e9cb34.png)

The fix is just getting rid of our first batch norm layer in `create_head`

**Comment by muellerzr** [comment]
A better solution would be to check for a batch norm layer at the end of the encoder

**Comment by jph00** [comment]
> The fix is just getting rid of our first batch norm layer in create_head

I don't think that works. I'm not entirely sure this is automatable. Perhaps the best we can hope for is a bool param that can be set via `model_meta`?
